components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go

```diff
@@ -462,6 +467,70 @@ func (nu *NodeUtilsImpl) calculateTotalInstanceCount() (int, error) {
 	return count, nil
 }
 
+func (nu *NodeUtilsImpl) restartPgNodes(leaderNode NodeIpHealth, pgIps []IP, infra *AutomateHAInfraDetails, statusSummary StatusSummary) error {
+	newSSHConfig := &SSHConfig{
+		sshUser:    infra.Outputs.SSHUser.Value,
+		sshPort:    infra.Outputs.SSHPort.Value,
+		sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+	}
+	sshUtil := NewSSHUtil(newSSHConfig)
+	// restart followers
+	nu.writer.Println("trying to restaring follower node")
+	restartFollowerNodeAndWaitForhealthy(leaderNode, pgIps, infra, statusSummary, sshUtil, nu.writer)
+	//restart leader
+	nu.writer.Println("trying to restaring leader node")
+	for _, pgIp := range pgIps {
+		nu.writer.Println("looking for leader node to restart")
+		if strings.EqualFold(pgIp.IP, leaderNode.IP) {
+			nu.writer.Println("Restaring leader node")
+			sshUtil.getSSHConfig().hostIP = pgIp.IP
+			res, err := sshUtil.connectAndExecuteCommandOnRemote(RESTART_BACKEND_COMMAND, true)
+			if err != nil {
+				writer.Errorf("error in executing restart backend supervisor command")
+				return status.Wrapf(err, status.RestartDeploymentServiceError, "error in executing restart backend supervisor command %s", res)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func restartFollowerNodeAndWaitForhealthy(leaderNode NodeIpHealth, pgIps []IP, infra *AutomateHAInfraDetails, statusSummary StatusSummary, sshUtil SSHUtil, writer *cli.Writer) error {
+	for _, pgIp := range pgIps {
+		if !strings.EqualFold(pgIp.IP, leaderNode.IP) {
+			sshUtil.getSSHConfig().hostIP = pgIp.IP
+			res, err := sshUtil.connectAndExecuteCommandOnRemote(RESTART_BACKEND_COMMAND, true)
+			if err != nil {
+				writer.Errorf("error in executing restart backend supervisor command")
+				return status.Wrapf(err, status.RestartDeploymentServiceError, "error in executing restart backend supervisor command %s", res)
+			}
+		}
+	}
+	writer.Println("polling follower node health")
+	followerHealth := statusSummary.GetPGFollwerNodes()
+	start := time.Now()
+	for {
+		healthy := false
+		for k, v := range followerHealth {
+			if v == SERVICE_HEALTH_OK {
+				writer.Println(fmt.Sprintf("follower node %s health %s", k, v))
+				healthy = true
+			}
+		}
+		if healthy {
+			writer.Println("all pg follower nodes are healthy")
+			break
+		}
+		time.Sleep(time.Second * SERVICE_HEALTH_CHECK_INTERVAL)
+		timeElaspsed := time.Since(start)
```

Comment:
either we can keep this outside the loop 
OR
use directly `time.Since(start).Seconds()